src/v0.rs

```diff
@@ -1281,6 +1281,12 @@ mod tests {
         t_nohash!("_RNvC6_123foo3bar", "123foo::bar");
     }
 
+    #[test]
+    fn demangle_crate_with_zero_disambiguator() {
+        t!("_RC4f128", "f128");
+        t_nohash!("_RC4f128", "f128");
```

Comment:
Do these tests fail without the change? When I was trying to work on this it seemed like these simple tests wouldn't emit `[0]` even as-is, and I was struggling to write a test that replicates the behavior from https://github.com/rust-lang/rust/pull/123816#discussion_r1573370985. Hence my messy branch at https://github.com/tgross35/rustc-demangle/blob/b29eead5e3337030843b978736748a3f1c142416/src/v0.rs#L1284-L1295